C/COSMA/build_tarballs.jl

```diff
@@ -53,37 +70,63 @@ cmake ../cosma \
     -DBUILD_SHARED_LIBS=ON \
     -DCOSMA_BLAS=OPENBLAS \
     -DCOSMA_SCALAPACK=OFF \
-    -DMPI_CXX_COMPILER=$bindir/mpicxx \
     -DMPI_C_COMPILER=$bindir/mpicc \
+    -DMPI_CXX_COMPILER=$bindir/mpicxx \
     -DOPENBLAS_LIBRARIES=$BLAS_LAPACK_LIB \
     -DOPENBLAS_INCLUDE_DIR=$includedir \
-    $OPENMP_CMAKE_FLAGS
+    $OPENMP_CMAKE_FLAGS \
+    $MPI_CMAKE_FLAGS
 
 make -j${nproc}
 make install
 
-install_license ../cosma/LICENCE
+install_license ../COSMA-*/LICENCE
+"""
+
+augment_platform_block = """
+    using Base.BinaryPlatforms
+    $(MPI.augment)
+    augment_platform!(platform::Platform) = augment_mpi!(platform)
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = supported_platforms()
 platforms = expand_cxxstring_abis(platforms)
+
+# MPItrampoline is not supported on `libgfortran3`. We need to expand
+# the gfortran versions here so that we can exclude it. Otherwise,
+# BinaryBuilder might choose `libgfortran3`, and the build will fail
+# since no MPI implementation is available.
+platforms = expand_gfortran_versions(platforms)
+
 filter!(!Sys.iswindows, platforms)
 
+# OpenMP is not supported. The work-around for `apple` above does not work for `aarch64`.
+filter!(p -> !(arch(p) == "aarch64" && Sys.isapple(p)), platforms)
 
 # The products that we will ensure are always built
 products = [
-    LibraryProduct("libgrid2grid", :grid2grid),
-    LibraryProduct("libcosma", :cosma)
+    # `libgrid2grid` existed in v2.2.0, but not any more in v2.5.1
+    # LibraryProduct("libgrid2grid", :grid2grid),
+    LibraryProduct("libcosma", :cosma),
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency(PackageSpec(name="OpenBLAS_jll", uuid="4536629a-c528-5b80-bd46-f80d51c5b363")),
-    Dependency(PackageSpec(name="MPICH_jll", uuid="7cb0a576-ebde-5e09-9194-50597f1243b4")),
     Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
 ]
 
+platforms, platform_dependencies = MPI.augment_platforms(platforms)
+# Avoid platforms where the MPI implementation isn't supported
+# OpenMPI
+platforms = filter(p -> !(p["mpi"] == "openmpi" && arch(p) == "armv6l" && libc(p) == "glibc"), platforms)
+# MPItrampoline
+platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && (Sys.iswindows(p) || libc(p) == "musl")), platforms)
+platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && Sys.isfreebsd(p)), platforms)
+append!(dependencies, platform_dependencies)
+
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"7.1.0")
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
+               augment_platform_block, julia_compat="1.6", preferred_gcc_version = v"7.1.0")
```

Comment:
If you assume that OpenBLAS_jll on aarch64 does ILP64, then you must require Julia v1.7

Reply:
Yes it does. Will do.